org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hyperlinking/SingleHoverShowingHyperlinkPresenter.java

```diff
@@ -55,37 +56,38 @@ public SingleHoverShowingHyperlinkPresenter(MultipleHyperlinkPresenter delegate)
 
 	@Override
 	public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
-		if (method.getName().startsWith("showHyperlinks") && (args.length >= 1) && (args[0] instanceof IHyperlink[])) {
-			List<IHyperlink> nullsafe = makeNullsafe(Arrays.asList((IHyperlink[]) args[0]));
-			if (!nullsafe.isEmpty()) {
+		if (method.getName().startsWith("showHyperlinks") && args.length >= 1 && args[0] instanceof IHyperlink[]) {
+			IHyperlink[] nullsafe = this.makeNullsafe((IHyperlink[]) args[0]);
+			if (nullsafe.length > 0) {
 				args[0] = nullsafe;
-				Object result = method.invoke(delegate, args);
-				List<IHyperlink> activeHyperlinks = nullsafe;
-				if (activeHyperlinks.size() == 1) {
-					IHyperlink singleHyperlink = activeHyperlinks.get(0);
-					if (Objects.equals(SHOW_ALWAYS, singleHyperlink.getTypeLabel())) {
+				Object result = method.invoke(this.delegate, args);
+				IHyperlink[] activeHyperlinks = nullsafe;
+				if (activeHyperlinks.length == 1) {
+					IHyperlink singleHyperlink = activeHyperlinks[0];
+					if (SingleHoverShowingHyperlinkPresenter.SHOW_ALWAYS.equals(singleHyperlink.getTypeLabel())) {
 						int start = singleHyperlink.getHyperlinkRegion().getOffset();
 						int end = start + singleHyperlink.getHyperlinkRegion().getLength();
 						reflect.set(delegate, "fSubjectRegion", new Region(start, end - start));
-						reflect.set(delegate, "fCursorOffset",
-								Integer.valueOf(JFaceTextUtil.getOffsetForCursorLocation(reflect.get(delegate, "fTextViewer"))));
+						reflect.set(delegate, "fCursorOffset", Integer.valueOf(
+								JFaceTextUtil.getOffsetForCursorLocation(reflect.get(delegate, "fTextViewer"))));
 						((AbstractInformationControlManager) reflect.get(delegate, "fManager")).showInformation();
 					}
 				}
 				return result;
 			}
-			return null; // showHyperlinks returns void 
+			return null;
 		}
-		return method.invoke(delegate, args);
+		return method.invoke(this.delegate, args);
+
 	}
 
 	/**
 	 * @since 2.9
 	 */
-	protected List<IHyperlink> makeNullsafe(List<IHyperlink> hyperlinks) {
-		if (hyperlinks.stream().anyMatch(hyperLink -> (hyperLink == null || hyperLink.getHyperlinkRegion() == null))) {
+	protected IHyperlink[] makeNullsafe(IHyperlink[] hyperlinks) {
+		if (Arrays.stream(hyperlinks).anyMatch(hyperLink -> (hyperLink == null || hyperLink.getHyperlinkRegion() == null))) {
 			List<IHyperlink> nullSafeList = new ArrayList<>();
-			hyperlinks.forEach(hyperLink -> {
+			Arrays.stream(hyperlinks).forEach(hyperLink -> {
```

Comment:
I'm still not a big fan of streams where the forEach modified some captured instance, e.g. the nullSafeList.
A old-school for loop may have been the better choice.
Since it looks correct though, I'll approve